### PR TITLE
Reuse Question 2 Logic for Channel Island, Content Updates Outcome 14a et al

### PR DESCRIPTION
**Pull request by ikennaokpala**
Supersedes #2574.
[Trello Card](https://trello.com/c/LrqmkiU5/149-check-if-you-need-a-uk-visa-new-transit-visa-logic-and-content)
## Description

This PR builds on top of the work done on on PR #2574 and #2554

This PR contains content update to the border control question.

A third new option has been introduced to clarify the transit rules that apply.

For users travelling to the Channel island or isle of man they will be presented with purpose of visit question without the transit option.

The logic here remains the same with purpose of visit (i.e Question 2).

Within, I have moved the the blocks and logic within the purpose of visit multiple choice question to the travel_response_next_route method (an instance method on CheckUkVisaFlow).

Upon request from the Content Team [via this logic document](https://docs.google.com/document/d/1HvldJwcU-ylFkCu8p4DDPxlE3of-bVKU-bCuAZ5kvz8/edit?ts=5767e9ab).

[Main Preview Link](https://smart-answers-pr-2594.herokuapp.com/check-uk-visa/y)
[GOV.UK](https://gov.uk/check-uk-visa/y)
## Factcheck Question 2a

[Preview Link](https://smart-answers-pr-2594.herokuapp.com/check-uk-visa/y/afghanistan/transit)
[GOV.UK](https://gov.uk/check-uk-visa/y/afghanistan/transit)
### Expected changes
- Change optiions for Question (2a), replacing YES/NO with the following
  - Channel Islands or Isle of man
  - Republic of Ireland
  - Somewhere else
### Before

screen shot 2016-05-27 at 16 55 01
### After

screen shot 2016-06-23 at 12 54 39
## Factcheck New Question(2b)

[Preview Link](https://smart-answers-pr-2594.herokuapp.com/check-uk-visa/y/afghanistan/transit/channel_islands_or_isle_of_man)
[GOV.UK](https://gov.uk/check-uk-visa/y/afghanistan/transit/channel_islands_or_isle_of_man)
### Expected changes
- Added new Question (2b) that reuses the same responses and next node logic with the purpose of visit question but doesn't contain the transit option
### Before

screen shot 2016-07-11 at 17 22 30
### After

screen shot 2016-06-23 at 12 56 00
## Factcheck New Outcome 14a

[Preview Link](https://smart-answers-pr-2594.herokuapp.com/check-uk-visa/y/syria/transit/republic_of_ireland)
[GOV.UK](https://gov.uk/check-uk-visa/y/syria/transit/republic_of_ireland)
### Expected changes
- Content update to outcome (14a)
### Before

screen shot 2016-05-27 at 16 56 34
### After

screen shot 2016-06-23 at 13 08 31

### CODE REVIEW

File: lib/smart_answer_flows/check-uk-visa.rb [hunk 1]
```diff
@@ -56,78 +57,42 @@ def define
...
+
+      #Q2a
+      multiple_choice :travelling_to_cta? do
+        option :channel_islands_or_isle_of_man
+        option :republic_of_ireland
+        option :somewhere_else
+
         on_response do |response|
-          calculator.purpose_of_visit_answer = response
+          calculator.travelling_to_cta_answer = response
```
Comment (leenagupte) [inline lib/smart_answer_flows/check-uk-visa.rb]: CTA should be defined somewhere, even if it's just in a comment.

**Comment by leenagupte** [comment]
@ikennaokpala , the end result of the code looks good.
However, I have a few comments about how you got there:

I don't think it's necessary to show the different iterations the new question (2a) went through before the department decided on the version it wanted to go live.

The original plan for question 2a (yes/no question) has never been live, so it doesn't need to be part of our commit history.

To that end, I would:

1. Squash commits 558c3af and 76526f2 so only the 3-point radio button version of question 2a survives.
2. Move the refactoring of the `purpose of visit` question that has been included in commit 76526f2 to a separate commit. The refactor affects a question that otherwise wouldn't have been affected by this story and it should be separated to make that clear.

Also, I think you should also update the PR description to just mention the changes that were finally agreed upon by the department, and point to that version of the logic document if you want.

I understand that this story in particular has a very tight deadline, so take these comments as optional.

I purposefully haven't commented on the 'further content changes', commit 	0418ac1 and later, because I was involved in them. @pmanrubia, do you want to look them over?

**Comment by ikennaokpala** [comment]
@leenagupte sure I will follow your advice and squash this, once the changes are final.
I prefer this finally before merge to help me track how this PR has evolved.

I will also at that point update the desciption to make the final content and logic changes, route and purpose of the PR.

**Comment by leenagupte** [comment]
@ikennaokpala Waiting until all of the changes have been finalised before squashing the commits works for me.

File: lib/smart_answer/calculators/uk_visa_calculator.rb [hunk 2]
```diff
@@ -106,6 +107,18 @@ def passing_through_uk_border_control?
...
+    end
+
+    def travelling_to_ireland?
+      @travelling_to_cta_answer == 'republic_of_ireland'
+    end
+
+    def travelling_to_elsewhere?
+      @travelling_to_cta_answer == 'somewhere_else'
+    end
+
```
Comment (ikennaokpala) [inline lib/smart_answer/calculators/uk_visa_calculator.rb]: @leenagupte no.. I did not give any thought to the approach you have just mentioned.
I felt at the time that these predicate methods were sufficient enough.

File: lib/smart_answer_flows/check-uk-visa.rb (hunk 1 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/check-uk-visa.rb]: It appears the git diff is a bit misleading

File: lib/smart_answer_flows/check-uk-visa.rb [hunk 3]
```diff
@@ -130,6 +130,40 @@ def define
...
+        option :republic_of_ireland
+        option :somewhere_else
+
+        on_response do |response|
+          calculator.travelling_to_cta_answer = response
+        end
+
+        next_node do
+          if calculator.travelling_to_channel_islands_or_isle_of_man?
+            next outcome(:channel_islands_or_isle_of_man?)
```
Comment (leenagupte) [inline lib/smart_answer_flows/check-uk-visa.rb]: This should be `next question(:channel_islands_or_isle_of_man?)`

**Comment by chrisroos** [comment]
What's the state of this PR, @ikennaokpala? Does it need the "Waiting on factcheck" label or is it just waiting to be merged?

**Comment by pmanrubia** [comment]
Thank you for the follow-up @chrisroos. I have added the `needs fact-check` label as it is currently `in progress`.

**Comment by ikennaokpala** [comment]
@chrisroos it has come back from factcheck and I am working on it at the moment

**Comment by ikennaokpala** [comment]
Finally passed factcheck.. 👍  🎉 🎉 🎉

**Pull request merged**